experimental/packages/opentelemetry-instrumentation/src/platform/node/instrumentation.ts

```diff
@@ -68,6 +72,90 @@ export abstract class InstrumentationBase<T = any>
     }
   }
 
+  protected override _wrap: typeof shimmer.wrap = (
+    moduleExports,
+    name,
+    wrapper
+  ) => {
+    if (!util.types.isProxy(moduleExports)) {
+      return shimmer.wrap(moduleExports, name, wrapper);
+    } else {
+      return this._wrapEsm(moduleExports, name, wrapper);
+    }
+  };
+
+  protected override _unwrap: typeof shimmer.unwrap = (moduleExports, name) => {
+    if (!util.types.isProxy(moduleExports)) {
+      return shimmer.unwrap(moduleExports, name);
+    } else {
+      return this._unwrapEsm(moduleExports, name);
+    }
+  };
+
+  private _wrapEsm: typeof shimmer.wrap = (moduleExports, name, wrapper) => {
```

Comment:
General comment --- non blocking.
As this function is only used once, have it inlined (rather than a private) would be better (IMHO) -- primarily for bundle size. although as a node only module thats not as necessary. But it might also contribute to refactoring so to minimize the number of calls. Like perhaps only have a single call shimmer.unwrap for example.